tokio-executor/src/threadpool/pool/mod.rs

```diff
@@ -402,6 +401,9 @@ impl Pool {
         use super::worker::Lifecycle::Signaled;
 
         if let Some((idx, worker_state)) = self.sleep_stack.pop(&self.workers, Signaled, false) {
+            let span = trace_span!("signal_work", idx);
```

Comment:
why not `let _enter = trace_spawn!("signal_work, idx).enter();`?

Reply:
@LucioFranco `Span::enter` borrows the span, so we can't do that. There has to be a `let` binding for the span that's entered or else the borrow will outlive the span. 

There's been some discussion of improving ergonomics in cases where the span is only entered a single time, see https://github.com/tokio-rs/tracing/issues/192 and https://github.com/tokio-rs/tracing/issues/158.

Reply:
@LucioFranco `Span::enter` borrows the span, so we can't do that. There has to be a `let` binding for the span that's entered or else the borrow will outlive the span. 

There's been some discussion of improving ergonomics in cases where the span is only entered a single time, see https://github.com/tokio-rs/tracing/issues/192 and https://github.com/tokio-rs/tracing/issues/158.